assets/js/search.js

```diff
@@ -39,62 +39,66 @@
   /**
    * Function displays the total items displayed on the page
    */
-  function showItemsCount(searchEntry, totalCount, numSubmodules) {
-    if (searchEntry.type === 'libraryEntries') {
+  function showItemsCount(totalCount, numSubmodules) {
+    if ($('.table-clickable-row'.length > 0)) {
       $('#search-results-count').show();
       if (totalCount > 0 && numSubmodules > 0) {
-        $('#search-results-count').html("<strong>" + totalCount + "</strong> repos (~<strong>" + numSubmodules + "</strong> modules)");
+        $('#search-results-count').html("<strong>" + totalCount + "</strong>repos(~<strong>" + numSubmodules + "</strong> modules)");
       } else {
         $('#search-results-count').text("0 repos");
       }
-    } else {
-      if (totalCount > 0 ) {
-        $('#search-results-count').html("<strong>" + totalCount + "</strong> post(s) found");
-      }
     }
   }
 
   /**
    * Function to display all items on the page
    */
   function showAllItems() {
-    $('.guide-card').show();
-    $('.category-head').show();
-    $('.categories ul li').show();
+    return (
+      $('.guide-card').show() &&
+      $('.category-head').show() &&
+      $('.categories ul li').show()
+    )
+  }
+
+
+  function initialDisplay() {
+    return (
+      $('#guide-listings').show() &&
+      $('#no-azure-results').hide() &&
+      $('#no-matches').hide()
+    );
+  }
+
+  /**
+   * Function to hide items on the page
+   */
+  function hideItems() {
+    return (
+      $('.guide-card').hide() &&
+      $('.category-head').hide() &&
+      $('.categories ul li').hide()
+    )
   }
 
   /**
    * Function that counts how many items are on the page
    */
   function showInitialItemsCount(searchEntry) {
     let numSubmodules = 0;
-    for (let i = 0; i < searchEntry.entries.length; i++) {
-      numSubmodules += libraryEntries[i].num_submodules;
+    for (let i = 0; i < searchEntry.length; i++) {
+      numSubmodules += window.libraryEntries[i].num_submodules;
     }
-    showItemsCount(searchEntry, searchEntry.entries.length, numSubmodules);
+    showItemsCount(searchEntry.length, numSubmodules);
   }
 
   /**
    * Function that where the search is being performed from
    */
   function detectSearchEntry() {
     let entries = [];
-    if (window.libraryEntries) {
-
-      entries = window.libraryEntries;
-      return {
-        entries,
-        type: 'libraryEntries'
-      };
-
-    } else if (window.guideEntries) {
-
-      entries = window.guideEntries;
-      return {
-        entries,
-        type: 'guideEntries'
-      };
-    }
+    entries = window.libraryEntries ? window.libraryEntries : window.guideEntries;
+    return entries;
```

Comment:
You can collapse these three lines into one: `return window.libraryEntries ? ... : ...;`